src/main/scala/org/apache/spark/ml/ModelBuilderSSP.scala

```diff
@@ -11,18 +11,24 @@ import org.apache.spark.mllib.tree.impurity.ImpurityCalculator
 /**
  * Helper for creating MLlib models which have private constructors.
  */
-object ModelBuilder {
+object ModelBuilderSSP {
 
   def newLogisticRegressionModel(
       coefficients: Vector,
       intercept: Double): LogisticRegressionModel = {
     new LogisticRegressionModel("lr", coefficients, intercept)
+      .setThreshold(.5)
+
   }
 
   def newLinearRegressionModel(
       coefficients: Vector,
       intercept: Double): LinearRegressionModel = {
-    new LinearRegressionModel("linr", coefficients, intercept)
+    val model = new LinearRegressionModel("linr", coefficients, intercept)
+    if (model.hasParam("loss")) {
```

Comment:
Could we remove this `if` by updating `build.sbt` to build spark-sql-perf against Spark 2.3 instead of 2.2?

Reply:
@smurching Spark 2.3 is not out yet.  I would keep the `if` at least half a year more.  The TPC-* part of `spark-sql-perf` works fine built against 2.2.

Reply:
Gotcha, yeah that sounds good to me - thanks for clarifying!